plugins/serializers/msgpack/msgpack_test.go

```diff
@@ -0,0 +1,237 @@
+
+	assert.Equal(t, true, isIdentical(t, m, m2))
+}
+
+func TestSerializeMetricString(t *testing.T) {
+	now := time.Now()
+	tags := map[string]string{
+		"cpu": "cpu0",
+	}
+	fields := map[string]interface{}{
+		"usage_idle": "foobar",
+	}
+	m, err := metric.New("cpu", tags, fields, now)
+	assert.NoError(t, err)
+
+	s := Serializer{}
+	var buf []byte
+	buf, err = s.Serialize(m)
+	assert.NoError(t, err)
+
+	m2 := &Metric{}
+	left, err := m2.UnmarshalMsg(buf)
+	assert.NoError(t, err)
+
+	assert.Equal(t, len(left), 0)
+
+	assert.Equal(t, true, isIdentical(t, m, m2))
+}
+
+func TestSerializeMultiFields(t *testing.T) {
+	now := time.Now()
+	tags := map[string]string{
+		"cpu": "cpu0",
+	}
+	fields := map[string]interface{}{
+		"usage_idle":  int64(90),
+		"usage_total": 8559615,
+	}
+	m, err := metric.New("cpu", tags, fields, now)
+	assert.NoError(t, err)
+
+	s := Serializer{}
+	var buf []byte
+	buf, err = s.Serialize(m)
+	assert.NoError(t, err)
+
+	m2 := &Metric{}
+	left, err := m2.UnmarshalMsg(buf)
+	assert.NoError(t, err)
+
+	assert.Equal(t, len(left), 0)
+
+	assert.Equal(t, true, isIdentical(t, m, m2))
+}
+
+func TestSerializeMetricWithEscapes(t *testing.T) {
+	now := time.Now()
+	tags := map[string]string{
+		"cpu tag": "cpu0",
+	}
+	fields := map[string]interface{}{
+		"U,age=Idle": int64(90),
+	}
+	m, err := metric.New("My CPU", tags, fields, now)
+	assert.NoError(t, err)
+
+	s := Serializer{}
+	var buf []byte
+	buf, err = s.Serialize(m)
+	assert.NoError(t, err)
+
+	m2 := &Metric{}
+	left, err := m2.UnmarshalMsg(buf)
+	assert.NoError(t, err)
+
+	assert.Equal(t, len(left), 0)
+
+	assert.Equal(t, true, isIdentical(t, m, m2))
+}
+
+func TestSerializeMultipleMetric(t *testing.T) {
+	now := time.Now()
+	tags := map[string]string{
+		"cpu tag": "cpu0",
+	}
+	fields := map[string]interface{}{
+		"U,age=Idle": int64(90),
+	}
+	m, err := metric.New("My CPU", tags, fields, now)
+	assert.NoError(t, err)
+
+	s := Serializer{}
+
+	encoded, err := s.Serialize(m)
+	assert.NoError(t, err)
+
+	// Multiple metrics in continous bytes stream
+	var buf []byte
+	buf = append(buf, encoded...)
+	buf = append(buf, encoded...)
+	buf = append(buf, encoded...)
+	buf = append(buf, encoded...)
+
+	left := buf
+	for len(left) > 0 {
+		decodeM := &Metric{}
+		left, err = decodeM.UnmarshalMsg(left)
+
+		assert.NoError(t, err)
+		assert.Equal(t, true, isIdentical(t, m, decodeM))
+	}
+}
+
+func TestSerializeBatch(t *testing.T) {
+	now := time.Now()
+	tags := map[string]string{
+		"cpu tag": "cpu0",
+	}
+	fields := map[string]interface{}{
+		"U,age=Idle": int64(90),
+	}
+	m, err := metric.New("My CPU", tags, fields, now)
+	assert.NoError(t, err)
+
+	metrics := []telegraf.Metric{m, m, m, m}
+
+	s := Serializer{}
+
+	buf, err := s.SerializeBatch(metrics)
+	assert.NoError(t, err)
+
+	left := buf
+	for len(left) > 0 {
+		decodeM := &Metric{}
+		left, err = decodeM.UnmarshalMsg(left)
+
+		assert.NoError(t, err)
+		assert.Equal(t, true, isIdentical(t, m, decodeM))
+	}
+}
```

Comment:
Could you please also add one test with a time set to something not equal zero!?

Reply:
Could you please explain the request a little bit?

Reply:
Oh man I missed the `now` in the end... :-S So please forget about the comment above. :-/ Anyway, sometimes we have problems with different time resolutions across OSes, so it might be easier for testing when using a fixed time. You decide.

Reply:
I already changed it to use `testutil.TestMetric` so I assume I can resolve this.